apps/api/v2/src/modules/oauth/oauth-client.controller.ts

```diff
@@ -13,25 +14,22 @@ import {
   HttpCode,
   HttpStatus,
   Logger,
-  Res,
+  UseGuards,
 } from "@nestjs/common";
 
 @Controller({
   path: "oauth-clients",
   version: "2",
 })
+@UseGuards(NextAuthGuard)
 export class OAuthClientController {
   private readonly logger = new Logger("OAuthClientController");
 
   constructor(private readonly oauthClientRepository: OAuthClientRepository) {}
 
   @Post("/")
   @HttpCode(HttpStatus.CREATED)
-  async createOAuthClient(
-    @Res({ passthrough: true }) res: Response,
-    @Body() createOAuthClientDto: CreateOAuthClientDto
-  ) {
-    const userId = res.locals.apiKey?.userId;
+  async createOAuthClient(@GetUser("id") userId: number, @Body() createOAuthClientDto: CreateOAuthClientDto) {
```

Comment:
@exception if we used the same approach for the apiKey, would we still need to set the apiKey in response.locals in ApiKeyService ?

it seems like we can completely avoid using response.locals following the approach in this pr?

Reply:
Yes that should be fine!